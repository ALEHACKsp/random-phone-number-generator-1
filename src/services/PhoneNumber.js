import Storage from './Storage';

class PhoneNumber {
  constructor() {
    this.storage = new Storage();
  }

  generatePhoneNumber() {
    return '0' + Math.floor(Math.random() * 900000000 + 100000000)
  }

  generate(count) {
    const existingCount = this.storage.count();
    const newCount = count + existingCount;
    for (let i = 0; i < count; i++) {
      this.storage.add(this.generatePhoneNumber());
    }
    const availableCount = this.storage.count();
    if (availableCount < newCount) {
      this.generate(newCount - availableCount);
    }
  }
}

export default new PhoneNumber();
