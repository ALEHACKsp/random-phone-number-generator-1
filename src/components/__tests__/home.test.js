import React from 'react';
import enzyme from 'enzyme';
import Home from '../home';
import PhoneNumber from 'services/PhoneNumber';

const phoneNumber = new PhoneNumber();
describe('Home Component', () => {
  let instance;
  let node;
  const props = {
    handleInputChange: jest.fn(),
    handlePageChange: jest.fn(),
    updatePhoneNumbers: jest.fn(),
    generatePhones: jest.fn()
  };

  const updatePhoneNumbers = jest.fn();
  beforeEach(() => {
    node = enzyme.shallow(<Home {...props} onClick={updatePhoneNumbers} />);
    instance = node.instance();
  });

  it('should render properly', () => {
    expect(node).toMatchSnapshot();
  });

  it('handles input changes', () => {
    expect(
      node.find('[name="number"]').simulate('change', { target: { name: 'number', value: 6 } })
    );
  });

  it('should throw an error if no number is input', () => {
    node.setState({ number: -1 });
    instance.generatePhones();
    const state = node.state();
    expect(state.number).toEqual(-1);
    expect(state.error).toBe(true);
    expect(state.message).toBe('The number entered exceeds/below the accepted limit');
  });

  it('should call the generate method if all is fine', () => {
    node.setState({ number: 20 });
    instance.generatePhones();
    const state = node.state();
    expect(state.number).toEqual(20);
  });

  it('should update stats for minimum and maximum', () => {
    node.setState({ phoneNumbers: ['0987609876', '0987654398'] });
    instance.getStatistics();
    const state = node.state();
    expect(state.total).toBe(2);
    expect(state.min).toBeDefined();
    expect(state.max).toBeDefined();
  });

  it('should test sorting PhoneNumbers', () => {
    node.setState({ sorter: 'desc', phoneNumbers: ['0787609876', '0787654389'] });
    instance.sortPhoneNumbers();
    const state = node.state();
    expect(state.phoneNumbers.length).toBeGreaterThan(1);
    expect(state.sorter).toBe('desc');
  });

  it('should call sortPhoneNumbers function', () => {
    node.setState({ phoneNumbers: ['0787609876', '0787654389'] });
    const event = {
      preventDefault: () => {},
      target: {
        value: 'asc'
      }
    };
    let spy;
    spy = jest.spyOn(instance, 'onSortChange');
    instance.onSortChange(event);
    const state = node.state();
    expect(state.sorter).toBe('asc');
    expect(spy).toHaveBeenCalled();
    spy.mockRestore();
  });

  it('generates phones numbers when the button is clicked', () => {
    const generatePhoneSpy = jest.spyOn(node.instance(), 'updatePhoneNumbers');
    node
      .find('.my-button')
      .at(0)
      .simulate('click');
    expect(generatePhoneSpy).toHaveBeenCalled();
  });
});
