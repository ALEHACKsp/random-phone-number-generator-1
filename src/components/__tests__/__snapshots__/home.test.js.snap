// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    generatePhones={[MockFunction]}
    handleInputChange={[MockFunction]}
    handlePageChange={[MockFunction]}
    onClick={[MockFunction]}
    updatePhoneNumbers={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="input-group"
        >
          <input
            className="form-control"
            name="number"
            onChange={[Function]}
            placeholder="You can only generate 10000 phone numbers, so as not to exceed the limit"
            type="number"
            value=""
          />
          <div
            className="input-group-append"
          >
            <button
              className="btn my-button"
              onClick={[Function]}
            >
              Generate Phone Numbers
            </button>
          </div>
        </section>,
        <Error />,
        <section>
          List of phone numbers
          <div
            className="card"
          >
            <div
              className="card-body card-table"
            >
              <div
                className="card-title in-line"
              >
                <Sorter
                  onChange={[Function]}
                  phoneNumbers={Array []}
                />
              </div>
              <div
                className="stats"
              >
                <MinMaxPhoneNumber
                  max={null}
                  min={null}
                  phoneNumbers={Array []}
                  total={0}
                />
              </div>
              <table
                className="table table-striped table-bordered table-sm"
              >
                <thead>
                  <tr>
                    <th>
                      Number
                    </th>
                    <th>
                      Phone Number
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <span
                    style={
                      Object {
                        "padding": "10px",
                      }
                    }
                  >
                    No phone numbers yet, generate some
                  </span>
                </tbody>
              </table>
            </div>
          </div>
        </section>,
      ],
      "className": "container mt-3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="form-control"
              name="number"
              onChange={[Function]}
              placeholder="You can only generate 10000 phone numbers, so as not to exceed the limit"
              type="number"
              value=""
            />,
            <div
              className="input-group-append"
            >
              <button
                className="btn my-button"
                onClick={[Function]}
              >
                Generate Phone Numbers
              </button>
            </div>,
          ],
          "className": "input-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "name": "number",
              "onChange": [Function],
              "placeholder": "You can only generate 10000 phone numbers, so as not to exceed the limit",
              "type": "number",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn my-button"
                onClick={[Function]}
              >
                Generate Phone Numbers
              </button>,
              "className": "input-group-append",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Generate Phone Numbers",
                "className": "btn my-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Generate Phone Numbers",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "error": undefined,
          "message": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "List of phone numbers",
            <div
              className="card"
            >
              <div
                className="card-body card-table"
              >
                <div
                  className="card-title in-line"
                >
                  <Sorter
                    onChange={[Function]}
                    phoneNumbers={Array []}
                  />
                </div>
                <div
                  className="stats"
                >
                  <MinMaxPhoneNumber
                    max={null}
                    min={null}
                    phoneNumbers={Array []}
                    total={0}
                  />
                </div>
                <table
                  className="table table-striped table-bordered table-sm"
                >
                  <thead>
                    <tr>
                      <th>
                        Number
                      </th>
                      <th>
                        Phone Number
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <span
                      style={
                        Object {
                          "padding": "10px",
                        }
                      }
                    >
                      No phone numbers yet, generate some
                    </span>
                  </tbody>
                </table>
              </div>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "List of phone numbers",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="card-body card-table"
              >
                <div
                  className="card-title in-line"
                >
                  <Sorter
                    onChange={[Function]}
                    phoneNumbers={Array []}
                  />
                </div>
                <div
                  className="stats"
                >
                  <MinMaxPhoneNumber
                    max={null}
                    min={null}
                    phoneNumbers={Array []}
                    total={0}
                  />
                </div>
                <table
                  className="table table-striped table-bordered table-sm"
                >
                  <thead>
                    <tr>
                      <th>
                        Number
                      </th>
                      <th>
                        Phone Number
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <span
                      style={
                        Object {
                          "padding": "10px",
                        }
                      }
                    >
                      No phone numbers yet, generate some
                    </span>
                  </tbody>
                </table>
              </div>,
              "className": "card",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="card-title in-line"
                  >
                    <Sorter
                      onChange={[Function]}
                      phoneNumbers={Array []}
                    />
                  </div>,
                  <div
                    className="stats"
                  >
                    <MinMaxPhoneNumber
                      max={null}
                      min={null}
                      phoneNumbers={Array []}
                      total={0}
                    />
                  </div>,
                  <table
                    className="table table-striped table-bordered table-sm"
                  >
                    <thead>
                      <tr>
                        <th>
                          Number
                        </th>
                        <th>
                          Phone Number
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <span
                        style={
                          Object {
                            "padding": "10px",
                          }
                        }
                      >
                        No phone numbers yet, generate some
                      </span>
                    </tbody>
                  </table>,
                ],
                "className": "card-body card-table",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Sorter
                        onChange={[Function]}
                        phoneNumbers={Array []}
                      />,
                      null,
                    ],
                    "className": "card-title in-line",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "onChange": [Function],
                        "phoneNumbers": Array [],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    null,
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <MinMaxPhoneNumber
                      max={null}
                      min={null}
                      phoneNumbers={Array []}
                      total={0}
                    />,
                    "className": "stats",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "max": null,
                      "min": null,
                      "phoneNumbers": Array [],
                      "total": 0,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <thead>
                        <tr>
                          <th>
                            Number
                          </th>
                          <th>
                            Phone Number
                          </th>
                        </tr>
                      </thead>,
                      <tbody>
                        <span
                          style={
                            Object {
                              "padding": "10px",
                            }
                          }
                        >
                          No phone numbers yet, generate some
                        </span>
                      </tbody>,
                    ],
                    "className": "table table-striped table-bordered table-sm",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <tr>
                          <th>
                            Number
                          </th>
                          <th>
                            Phone Number
                          </th>
                        </tr>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <th>
                              Number
                            </th>,
                            <th>
                              Phone Number
                            </th>,
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Number",
                            },
                            "ref": null,
                            "rendered": "Number",
                            "type": "th",
                          },
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Phone Number",
                            },
                            "ref": null,
                            "rendered": "Phone Number",
                            "type": "th",
                          },
                        ],
                        "type": "tr",
                      },
                      "type": "thead",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <span
                          style={
                            Object {
                              "padding": "10px",
                            }
                          }
                        >
                          No phone numbers yet, generate some
                        </span>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "No phone numbers yet, generate some",
                          "style": Object {
                            "padding": "10px",
                          },
                        },
                        "ref": null,
                        "rendered": "No phone numbers yet, generate some",
                        "type": "span",
                      },
                      "type": "tbody",
                    },
                  ],
                  "type": "table",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="input-group"
          >
            <input
              className="form-control"
              name="number"
              onChange={[Function]}
              placeholder="You can only generate 10000 phone numbers, so as not to exceed the limit"
              type="number"
              value=""
            />
            <div
              className="input-group-append"
            >
              <button
                className="btn my-button"
                onClick={[Function]}
              >
                Generate Phone Numbers
              </button>
            </div>
          </section>,
          <Error />,
          <section>
            List of phone numbers
            <div
              className="card"
            >
              <div
                className="card-body card-table"
              >
                <div
                  className="card-title in-line"
                >
                  <Sorter
                    onChange={[Function]}
                    phoneNumbers={Array []}
                  />
                </div>
                <div
                  className="stats"
                >
                  <MinMaxPhoneNumber
                    max={null}
                    min={null}
                    phoneNumbers={Array []}
                    total={0}
                  />
                </div>
                <table
                  className="table table-striped table-bordered table-sm"
                >
                  <thead>
                    <tr>
                      <th>
                        Number
                      </th>
                      <th>
                        Phone Number
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <span
                      style={
                        Object {
                          "padding": "10px",
                        }
                      }
                    >
                      No phone numbers yet, generate some
                    </span>
                  </tbody>
                </table>
              </div>
            </div>
          </section>,
        ],
        "className": "container mt-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="form-control"
                name="number"
                onChange={[Function]}
                placeholder="You can only generate 10000 phone numbers, so as not to exceed the limit"
                type="number"
                value=""
              />,
              <div
                className="input-group-append"
              >
                <button
                  className="btn my-button"
                  onClick={[Function]}
                >
                  Generate Phone Numbers
                </button>
              </div>,
            ],
            "className": "input-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "name": "number",
                "onChange": [Function],
                "placeholder": "You can only generate 10000 phone numbers, so as not to exceed the limit",
                "type": "number",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn my-button"
                  onClick={[Function]}
                >
                  Generate Phone Numbers
                </button>,
                "className": "input-group-append",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Generate Phone Numbers",
                  "className": "btn my-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Generate Phone Numbers",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "error": undefined,
            "message": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "List of phone numbers",
              <div
                className="card"
              >
                <div
                  className="card-body card-table"
                >
                  <div
                    className="card-title in-line"
                  >
                    <Sorter
                      onChange={[Function]}
                      phoneNumbers={Array []}
                    />
                  </div>
                  <div
                    className="stats"
                  >
                    <MinMaxPhoneNumber
                      max={null}
                      min={null}
                      phoneNumbers={Array []}
                      total={0}
                    />
                  </div>
                  <table
                    className="table table-striped table-bordered table-sm"
                  >
                    <thead>
                      <tr>
                        <th>
                          Number
                        </th>
                        <th>
                          Phone Number
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <span
                        style={
                          Object {
                            "padding": "10px",
                          }
                        }
                      >
                        No phone numbers yet, generate some
                      </span>
                    </tbody>
                  </table>
                </div>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "List of phone numbers",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="card-body card-table"
                >
                  <div
                    className="card-title in-line"
                  >
                    <Sorter
                      onChange={[Function]}
                      phoneNumbers={Array []}
                    />
                  </div>
                  <div
                    className="stats"
                  >
                    <MinMaxPhoneNumber
                      max={null}
                      min={null}
                      phoneNumbers={Array []}
                      total={0}
                    />
                  </div>
                  <table
                    className="table table-striped table-bordered table-sm"
                  >
                    <thead>
                      <tr>
                        <th>
                          Number
                        </th>
                        <th>
                          Phone Number
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <span
                        style={
                          Object {
                            "padding": "10px",
                          }
                        }
                      >
                        No phone numbers yet, generate some
                      </span>
                    </tbody>
                  </table>
                </div>,
                "className": "card",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="card-title in-line"
                    >
                      <Sorter
                        onChange={[Function]}
                        phoneNumbers={Array []}
                      />
                    </div>,
                    <div
                      className="stats"
                    >
                      <MinMaxPhoneNumber
                        max={null}
                        min={null}
                        phoneNumbers={Array []}
                        total={0}
                      />
                    </div>,
                    <table
                      className="table table-striped table-bordered table-sm"
                    >
                      <thead>
                        <tr>
                          <th>
                            Number
                          </th>
                          <th>
                            Phone Number
                          </th>
                        </tr>
                      </thead>
                      <tbody>
                        <span
                          style={
                            Object {
                              "padding": "10px",
                            }
                          }
                        >
                          No phone numbers yet, generate some
                        </span>
                      </tbody>
                    </table>,
                  ],
                  "className": "card-body card-table",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <Sorter
                          onChange={[Function]}
                          phoneNumbers={Array []}
                        />,
                        null,
                      ],
                      "className": "card-title in-line",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "onChange": [Function],
                          "phoneNumbers": Array [],
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      null,
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <MinMaxPhoneNumber
                        max={null}
                        min={null}
                        phoneNumbers={Array []}
                        total={0}
                      />,
                      "className": "stats",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "max": null,
                        "min": null,
                        "phoneNumbers": Array [],
                        "total": 0,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <thead>
                          <tr>
                            <th>
                              Number
                            </th>
                            <th>
                              Phone Number
                            </th>
                          </tr>
                        </thead>,
                        <tbody>
                          <span
                            style={
                              Object {
                                "padding": "10px",
                              }
                            }
                          >
                            No phone numbers yet, generate some
                          </span>
                        </tbody>,
                      ],
                      "className": "table table-striped table-bordered table-sm",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <tr>
                            <th>
                              Number
                            </th>
                            <th>
                              Phone Number
                            </th>
                          </tr>,
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": Array [
                              <th>
                                Number
                              </th>,
                              <th>
                                Phone Number
                              </th>,
                            ],
                          },
                          "ref": null,
                          "rendered": Array [
                            Object {
                              "instance": null,
                              "key": undefined,
                              "nodeType": "host",
                              "props": Object {
                                "children": "Number",
                              },
                              "ref": null,
                              "rendered": "Number",
                              "type": "th",
                            },
                            Object {
                              "instance": null,
                              "key": undefined,
                              "nodeType": "host",
                              "props": Object {
                                "children": "Phone Number",
                              },
                              "ref": null,
                              "rendered": "Phone Number",
                              "type": "th",
                            },
                          ],
                          "type": "tr",
                        },
                        "type": "thead",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <span
                            style={
                              Object {
                                "padding": "10px",
                              }
                            }
                          >
                            No phone numbers yet, generate some
                          </span>,
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "No phone numbers yet, generate some",
                            "style": Object {
                              "padding": "10px",
                            },
                          },
                          "ref": null,
                          "rendered": "No phone numbers yet, generate some",
                          "type": "span",
                        },
                        "type": "tbody",
                      },
                    ],
                    "type": "table",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
